dynamic_programming/word_break.py

```diff
@@ -0,0 +1,25 @@
+def wordbreak(word: str, word_dict: list) -> bool:

```

Comment:
```suggestion
def wordbreak(word: str, word_dict: set[str]) -> bool:
```
There's no need to make `word_dict` a list and then immediately convert it to a set. Just because LeetCode set up the problem with `word_dict` as a list doesn't mean _we_ have to.